doc.js

```diff
@@ -220,6 +235,78 @@ define('doc', ['event'], function(event) {
 				return query(result);
 			},
 
+			'closest' : function(selector) {
+				// only works with non-composite selectors. e.g. '.class' or '#id' or 'div', but not: '#id .class' or 'ul.class'
+				var elements = [];
+				var selectorType;
+				var replacedSelector = selector;
+
+				if (matcher.isTag(selector)) {
+					selectorType = 'tag';
+				} else if (matcher.isClass(selector)) {
+					replacedSelector = selector.replace('.', '');
+					selectorType = 'class';
+				} else if (matcher.isId(selector)) {
+					replacedSelector = selector.replace('#', '');
+					selectorType = 'id';
+				} else {
+					console.warn('You are using composite selectors. e.g. "#id .class" or "tag.class". This selector will only work on Chrome 41+, Firefox 35+ or Opera 28+');
+				}
+
+				var checkForClosestParent = function(el) {
+					var element = el.first();
+					if (element) {
+						switch (selectorType) {
+							case 'tag':
+								return (replacedSelector.toUpperCase() === element.tagName)
+									? element
+									: checkForClosestParent(el.parent());
+								break;
+							case 'class':
+								return (el.hasClass(replacedSelector))
+									? element
+									: checkForClosestParent(el.parent());
+								break;
+							case 'id':
+								return (element.id === replacedSelector)
+									? element
+									: checkForClosestParent(el.parent());
+								break;
+							default:
+								throw new SyntaxError("You cannot use composite selector. e.g. 'tag.class' or '#id tag.class tag'. Use simple selectors like '#id', '.class' or 'tag'");
+								break;
+						}
+					}
+					return null;
+				};
+
+				this.each(function(el) {
+					if (el.closest === 'function') {
+						var element = el.first();
+						var closestElement = element.closest(selector);
+						if (closestElement) {
+							elemenets.push(el);
+						}
+					} else {
+						var closestElement = checkForClosestParent(query(el).parent());
+						if (closestElement) {
+							elements.push(closestElement);
+						}
+					}
+				});
+
+				for (var i = elements.length - 1; i > 0; i--) {
```

Comment:
this is not really a problem as nothing is computed. `elements.length` is just a property, not a function and is already cached.